### PR TITLE
implement token and http client functions

### PR DESCRIPTION
**Pull request by maaland**
# Checklist

- [x] `make test` passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](./doc/guides/commit-message.md#commit-message-guidelines)

## Description of change

This PR adds
* A function to generate a refreshable token that can be used to make authenticated requests, for instance to the Knowledge API
* A function to generate a client, authenticated with this refreshable token, which can be used to call the Knowledge API
* Some cleanup in relation to the above

When this is approved and merged, I will create tickets to implement the same functionality in the Node and Python SDKs.

### CODE REVIEW

File: oauth2/token.go [hunk 1]
```diff
@@ -16,55 +16,66 @@ package oauth2
...
-
-// Token returns a token or an error.
+// GetRefreshableTokenSource Token returns a token or an error.
 // Token must be safe for concurrent use by multiple goroutines.
 // The returned Token must not be modified.
-func (t *tokenSource) Token() (*oauth2.Token, error) {
-	resp, err := t.client.GetAccessToken(t.ctx, &identitypb.GetAccessTokenRequest{
-		// Id: "urn:indy:token",
-	}, grpc.PerRPCCredentials(oauth.TokenSource{TokenSource: t.applicationToken}))
+func GetRefreshableTokenSource(credentialsLoaders []config.CredentialsLoader) (oauth2.TokenSource, error) {
```
Comment (marvinkite) [inline oauth2/token.go:36]: Please add the `ctx context.Context` everywhere. This is a very powerful feature to have it. This was you can use custom http.Client which may be used in unit testing or orchestrated with custom wrapper.

File: oauth2/token.go [hunk 2]
```diff
@@ -16,55 +16,66 @@ package oauth2
...
-		}, nil
+
+	// refreshable token source, it can refresh every time we need a new one transparently.
+	return oauth2.ReuseTokenSource(token, tokenSource), nil
+}
+
+// GetHTTPClient returns an authenticated HTTP client
+// that always injects a valid token.
+func GetHTTPClient(credentialsLoaders []config.CredentialsLoader) (*http.Client, error) {
+	ctx := context.Background()
```
Comment (marvinkite) [inline oauth2/token.go:73]: Please add the `ctx context.Context` everywhere. This is a very powerful feature to have it. This was you can use custom http.Client which may be used in unit testing or orchestrated with custom wrapper.

File: oauth2/token.go (hunk 1 above)
Comment (maaland) [reply oauth2/token.go]: Good point. Added.

**Review by marvinkite — APPROVED** [review]

File: examples/oauth2/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Token Generation
+
+## Enviroment
+
+`INDYKITE_APPLICATION_CREDENTIALS_FILE`
```
Comment (arxeiss) [inline examples/oauth2/README.md:5]: `DefaultEnvironmentLoader` supports also `INDYKITE_APPLICATION_CREDENTIALS`
Where you specify credentials directly inside that Env var

**Pull request merged**